src/Collection/Collection.php

```diff
@@ -73,17 +73,29 @@ public function unserialize($collection)
     }
 
     /**
-     * Throws an exception.
+     * {@inheritDoc}
      *
-     * Issuing a count on a Collection can have many side effects, some making the
-     * Collection unusable after the count operation.
-     *
-     * @return void
-     * @throws \LogicException
+     * @return int
      */
     public function count()
     {
-        throw new LogicException('You cannot issue a count on a Collection.');
+        $traversable = $this->optimizeUnwrap();
+
+        if (is_array($traversable)) {
+            return count($traversable);
+        }
+
+        return iterator_count($traversable);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return int
+     */
+    public function countKeys()
```

Comment:
Should this maybe be `keyCount()`? as countKeys sounds more like exec and void command than a noun.

Reply:
not sure, `countKeys` sounds like the right choice to me.

Reply:
My thinking was:

count() is just a short form of getCount() and means the noun(), like getCollection() alias collection() etc.
So logically, the count for keys would be getKeyCount() or just keyCount().

But no big deal.

Reply:
That makes sense. I mostly wanted to keep the names close when autocompleting

Reply:
Good autocomplete will also find it when you start typing "key" IMO :)